paasta_tools/spark_tools.py

```diff
@@ -247,3 +248,23 @@ def get_spark_driver_monitoring_labels(
         "spark.yelp.com/driver_ui_port": ui_port_str,
     }
     return labels
+
+
+def get_spark_memory_in_unit(mem: str, unit: str) -> float:
+    """
+    Converts Spark memory to the desired unit.
+    mem is the same format as JVM memory strings: just number or number followed by 'k', 'm', 'g' or 't'.
+    unit can be 'k', 'm', 'g' or 't'.
+    Returns memory as an integer converted to the desired unit.
+    """
+    memory_bytes = 0.0
+    if mem:
+        if mem[-1] in MEM_MULTIPLIER:
+            memory_bytes = float(mem[:-1]) * MEM_MULTIPLIER[mem[-1]]
+        else:
+            try:
+                memory_bytes = float(mem)
+            except ValueError:
+                print(f"Unable to parse memory value {mem}")
+    memory_unit = memory_bytes / MEM_MULTIPLIER[unit]
```

Comment:
it's probably worth doing some validation in `paasta validate` / the schema to ensure that this is correctly specified: right now, if there's an issue with the value provided, this will return 0

(also, we probably want to verify that the provided `unit` is valid too - otherwise the division line will throw an exception when indexing into MEM_MULTIPLIER)

Reply:
(oh, i see - we hardcode the `unit` when we call this)

Reply:
Changed it to return a default 2 GB memory